internal/gcsx/file_cache_reader.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcsx
+
+import (
+	"github.com/googlecloudplatform/gcsfuse/v2/common"
+	"github.com/googlecloudplatform/gcsfuse/v2/internal/cache/file"
+	"github.com/googlecloudplatform/gcsfuse/v2/internal/storage/gcs"
+)
+
+type FileCacheReader struct {
+	Reader
+	obj    *gcs.MinObject
+	bucket gcs.Bucket
+
+	// fileCacheHandler is used to get file cache handle and read happens using that.
+	// This will be nil if the file cache is disabled.
+	fileCacheHandler *file.CacheHandler
+
+	// cacheFileForRangeRead is also valid for cache workflow, if true, object content
+	// will be downloaded for random reads as well too.
+	cacheFileForRangeRead bool
+
+	// fileCacheHandle is used to read from the cached location. It is created on the fly
+	// using fileCacheHandler for the given object and bucket.
+	fileCacheHandle *file.CacheHandle
+
+	metricHandle common.MetricHandle
+}
+
+func NewFileCacheReader(o *gcs.MinObject, bucket gcs.Bucket, fileCacheHandler *file.CacheHandler, cacheFileForRangeRead bool, metricHandle common.MetricHandle) FileCacheReader {
```

Comment:
This is little weird. you need to add the variables as and when they are required. But i guess this is okay since we are just moving around hte code.